cibw_before_build.sh

```diff
@@ -0,0 +1,22 @@
+python -m pip uninstall numpy
```

Comment:
Isn't this file the same as build.sh?

Reply:
@louisabraham  Yes, to a large extent. I should've been more clear in that the title "[WIP]" means "work-in-progress" and should not be merged because this PR has a `pyproject.yml` file which would likely be picked up by your use of `cibuildwheel` in your TravisCI process. Before it gets merged, I want to keep all of your files untouched (i.e., `setup.py`, `build.sh`, etc) so that you are still able to build and deploy if necessary. Right now with this merged, deployment is somewhat in limbo between TravisCI and Github Actions. I'm still working my way through all of this stuff so I'm fumbling as I go so, if you're cool with it, I'll reverse this merge first and create a new "[WIP]" where I can continue testing?